exist-core/src/main/java/org/exist/xquery/ForExpr.java

```diff
@@ -229,7 +229,10 @@ private void processItem(LocalVariable var, Item contextItem, Sequence in, Seque
         context.proceed(this);
         context.setContextSequencePosition(p, in);
         if (positionalVariable != null) {
-            at.setValue(new IntegerValue(this, p + 1));
+            if(contextItem.getType() == Type.ANY_ATOMIC_TYPE)
```

Comment:
@Robin6939 For formatting:

1. Insert a space between `if` and `(`
2. Add braces even for one line if statements.

e.g.:

```java
if (contextItem.getType() == Type.ANY_ATOMIC_TYPE) {
    at.setValue(new IntegerValue(this, 0));
} else {
    at.setValue(new IntegerValue(this, p + 1));
}
```

Reply:
how about 
```java
final int nextPosition = contextItem.getType() == Type.ANY_ATOMIC_TYPE ? 0 : p + 1;
at.setValue(new IntegerValue(this, nextPosition));
```